components/BottomSheet/src/MDCBottomSheetController.m

```diff
@@ -93,7 +93,12 @@ - (BOOL)accessibilityPerformEscape {
   [self
       dismissViewControllerAnimated:YES
                          completion:^{
-                           [weakSelf.delegate bottomSheetControllerDidDismissBottomSheet:weakSelf];
+                           __strong __typeof(weakSelf) strongSelf = weakSelf;
```

Comment:
This is a good improvement (switching to `__typeof`), but our team's convention is to avoid `typeof` and  `__typeof` whenever possible. We prefer to use explicitly-typed pointers in this circumstances.  I think in this case it's acceptable to use a pointer of type `MDCBottomSheetController *` explicitly.

If there's a specific reason not to use an explicit pointer type in this case, I think it's worth adding a comment to explain to future maintainers.